clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -328,6 +333,21 @@ void runOnce() {
         client.poll(pollTimeout, currentTimeMs);
     }
 
+    // We handle {@code TransactionalIdAuthorizationException} and {@code ClusterAuthorizationException} by first
+    // failing the inflight requests, then transition the state to UNINITIALIZED so that the user doesn't need to
+    // instantiate the producer again.
+    private boolean shouldHandleAuthorizationError(RuntimeException exception) {
```

Comment:
Just curious -- if we get an auth error on another request (ie, not initProducerId) do we expect to start over by initializing with a new ID? 

Also what is the goal with the poll call? Is it just replacing line 308? Would the code work without it?

Reply:
it seems like all of the non-initProducerId TransactionalIdAuthorizationException and ClusterAuthorizationException are fatal.

For the poll: I think we don't need it because there's no outbound request, as it should've been already polled in the previous `runOnce`.  The tests seem to work without so i'll remove it.

Reply:
I might have added it there because a few lines down, in the fatal error handling, the client was polled before return.